beacon-chain/blockchain/process_block.go

```diff
@@ -420,23 +421,83 @@ func (s *Service) savePostStateInfo(ctx context.Context, r [32]byte, b interface
 }
 
 // This removes the attestations in block `b` from the attestation mem pool.
-func (s *Service) pruneAttsFromPool(headBlock interfaces.ReadOnlySignedBeaconBlock) error {
+func (s *Service) pruneAttsFromPool(ctx context.Context, headState state.BeaconState, headBlock interfaces.ReadOnlySignedBeaconBlock) error {
 	atts := headBlock.Block().Body().Attestations()
 	for _, att := range atts {
-		if features.Get().EnableExperimentalAttestationPool {
-			if err := s.cfg.AttestationCache.DeleteCovered(att); err != nil {
-				return errors.Wrap(err, "could not delete attestation")
+		if !att.IsAggregated() {
+			if err := s.cfg.AttPool.DeleteUnaggregatedAttestation(att); err != nil {
+				return err
 			}
-		} else if att.IsAggregated() {
-			if err := s.cfg.AttPool.DeleteAggregatedAttestation(att); err != nil {
+			continue
+		}
+
+		if att.Version() == version.Phase0 {
+			if features.Get().EnableExperimentalAttestationPool {
+				if err := s.cfg.AttestationCache.DeleteCovered(att); err != nil {
+					return err
+				}
+			} else if err := s.cfg.AttPool.DeleteAggregatedAttestation(att); err != nil {
 				return err
 			}
-		} else {
-			if err := s.cfg.AttPool.DeleteUnaggregatedAttestation(att); err != nil {
+			continue
+		}
+
+		if err := s.pruneElectraAttsFromPool(ctx, headState, att); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// pruneElectraAttsFromPool handles removing aggregated Electra attestations from the pool after receiving a block.
+// Because in Electra block attestations can combine aggregates for multiple committees, comparing attestation bits
+// of a block attestation with attestations bits of an aggregate can cause unexpected results, leading to covered
+// aggregates not being removed from the pool.
+//
+// To make sure aggregates are removed, we decompose the block attestation into dummy aggregates, with each
+// aggregate accounting for one committee. This allows us to compare aggregates in the same way it's done for
+// Phase0. Even though we can't provide a valid signature for the dummy aggregate, it does not matter because
+// signatures play no part in pruning attestations.
+func (s *Service) pruneElectraAttsFromPool(ctx context.Context, headState state.BeaconState, att ethpb.Att) error {
+	if att.Version() == version.Phase0 {
+		log.Error("Called pruneElectraAttsFromPool with a Phase0 attestation")
+		return nil
+	}
+
+	offset := uint64(0)
+
+	committeeIndices := att.CommitteeBitsVal().BitIndices()
+	committees, err := helpers.AttestationCommittees(ctx, headState, att)
+	if err != nil {
+		return err
+	}
+
+	for i, c := range committees {
+		ab := bitfield.NewBitlist(uint64(len(c)))
+		for j := uint64(0); j < uint64(len(c)); j++ {
```

Comment:
is something like the below needed for safety? 
if offset+uint64(len(c)) > uint64(att.GetAggregationBits().Len()) {
    return errors.New("committee bits exceed available aggregation bits")
}

I think you can also do for j := range len(c), don't have a strong preference for it but it can be useful.

is it possible to have a test for the check I suggested?